src/pixelator/pixeldataset/utils.py

```diff
@@ -217,6 +219,76 @@ def write_anndata(adata: AnnData, filename: PathType) -> None:
     adata.write(filename=filename, compression="gzip")
 
 
+def _compute_sub_communities(
+    component_edgelist: pd.DataFrame, n_edges_reconnect: int | None = None
+):
+    component_edgelist = (
+        component_edgelist.groupby(["upia", "upib"], observed=True)["count"]
+        .count()
+        .reset_index()
+        .sort_values(["upia", "upib"])
+    )
+    edgelist_tuple = list(
+        map(tuple, np.array(component_edgelist[["upia", "upib", "count"]]))
+    )
+    component_communities_dict = leiden(
+        edgelist_tuple,
+        resolution=0.5 * LEIDEN_RESOLUTION,
+        random_seed=42,
+    )
+    component_communities = pd.Series(component_communities_dict)
+
+    return component_communities
+
+
+def _assess_doublet(component_edgelist: pd.DataFrame):
+    """Check whether a component is a potential doublet and how many edges should be removed to split it.
+
+    A component is a potential doublet if a) it has more than one community and
+    b) the second largest community is at least 20% of the size of the largest
+    community. (If the other communities are smaller they are assumed to be debries.)
+    """
+    component_communities = _compute_sub_communities(component_edgelist)
+    component_community_sizes = component_communities.value_counts().sort_values(
+        ascending=False
+    )
+    if len(component_community_sizes) > 1 and component_community_sizes.iloc[1] > (
+        0.2 * component_community_sizes.iloc[0]
+    ):
+        edges_to_remove = (
+            component_edgelist["upia"].map(component_communities)
+            != component_edgelist["upib"].map(component_communities)
+        ).sum()
+        return True, edges_to_remove
+    else:
+        return False, 0
+
+
+def mark_potential_doublets(
+    edgelist: pd.DataFrame,
+) -> pd.Series:
```

Comment:
```suggestion
) -> tuple[pd.Series, pd.Series]:
```
I'm a bit unsure here, but shouldn't this be the type annotation?